src/cfnlint/rules/resources/NoEcho.py

```diff
@@ -0,0 +1,47 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: MIT-0
+"""
+from cfnlint.helpers import bool_compare
+from cfnlint.rules import CloudFormationLintRule
+from cfnlint.rules import RuleMatch
+
+
+class NoEcho(CloudFormationLintRule):
+    id = 'W4002'
+    shortdesc = 'Check for NoEcho References'
+    description = 'Check if there is a NoEcho enabled parameter referenced within a resources Metadata section'
+    source_url = 'https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html#parameters-section-structure-properties'
+    tags = ['resources', 'NoEcho']
+
+    def match(self, cfn):
+        matches = []
+        no_echo_params = []
+        parameters = cfn.get_parameters()
+        for parameter_name, parameter_value in parameters.items():
+            noecho = parameter_value.get('NoEcho', default=False)
+            if bool_compare(noecho, True):
+                no_echo_params.append(parameter_name)
+
+        if not no_echo_params:
+            return no_echo_params
```

Comment:
@Birds are these lines necessary?
Wouldn't `[]` get returned either way?

Reply:
You're correct, this can certainly work without those lines - At the end of the day it'll be returned with an empty array either way